test/packages_test.rb

```diff
@@ -1016,4 +1016,85 @@ def product(properties = {})
       end
     end
   end
+
+  describe "#SelectProduct" do
+    before do
+      allow(subject).to receive(:Initialize).with(true)
+    end
+
+    context "when it is called in continue stage" do
+      before do
+        allow(Yast::Stage).to receive(:cont).and_return(true)
+      end
+
+      it "do not install any product" do
+        expect(Yast::Pkg).not_to receive(:ResolvableInstall)
+        subject.SelectProduct
+      end
+    end
+
+    context "when it is called in installation stage" do
```

Comment:
NP: That would mean `Yast::Stage.initial` call, but according to the mock below it actually is `when called outside the second stage`.